pandas/core/groupby.py

```diff
@@ -2315,6 +2315,11 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
                 # groupby
                 else:
                     cat = self.grouper.unique()
```

Comment:
ahh, ok for that case then (though try to use vectorized operations here). In fact probably create a method on ``Categorical``, call it ``._codes_for_sort(sort=True/False)`` where you can handle this in a method (rather than clutter it up here).